spec/mongo/transactions_examples_spec.rb

```diff
@@ -0,0 +1,236 @@
+                 Mongo::Error::NoServerAvailable => e
+
+            puts 'Transaction aborted. Caught exception during transaction.'
+            raise unless e.label?(Mongo::Session::TRANSIENT_TRANSACTION_ERROR_LABEL)
+
+            puts "#{Mongo::Session::TRANSIENT_TRANSACTION_ERROR_LABEL}, retrying transaction ..."
+            next
+          end
+        end
+      end
+
+      # End Transactions Retry Example 1
+
+      let(:run_transaction) do
+        run_transaction_with_retry(session) { |s| update_employee_info(s) }
+      end
+
+      it 'makes the changes to the database' do
+        run_transaction
+        employee = hr[:employees].find({ employee: 3 }, limit: 1).first
+        expect(employee).to_not be_nil
+        expect(employee['status']).to eq('Inactive')
+      end
+    end
+
+    context 'Transactions Retry Example 2' do
+
+      # Start Transactions Retry Example 2
+
+      def commit_with_retry(session)
+        loop do
+          begin
+            session.commit_transaction
+            puts 'Transaction committed.'
+            break
+          rescue Mongo::Error::OperationFailure, Mongo::Error::SocketError,
+                 Mongo::Error::NoServerAvailable => e
+
+            if e.label?(Mongo::Session::UNKNOWN_TRANSACTION_COMMIT_LABEL)
+              puts "#{Mongo::Session::UNKNOWN_TRANSACTION_COMMIT_LABEL}, retrying commit operation..."
+              next
+            else
+              puts 'Error during commit ...'
+              raise
+            end
+          end
+        end
+      end
+
+      # End Transactions Retry Example 2
+
+      let(:run_transaction) do
+        session.start_transaction
+        hr[:employees].insert_one({ employee: 4, status: 'Active' }, session: session)
+        reporting[:events].insert_one({ employee: 4, status: { new: 'Active', old: nil } },
+                                      session: session)
+        commit_with_retry(session)
+      end
+
+      it 'makes the changes to the database' do
+        run_transaction
+        employee = hr[:employees].find({ employee: 4 }, limit: 1).first
+        expect(employee).to_not be_nil
+        expect(employee['status']).to eq('Active')
+      end
+    end
+  end
+
+  context 'Transactions Retry Example 3 (combined example)' do
+
+    let(:client) do
+      authorized_client
+    end
+
+    let(:run_transaction) do
+
+      # Start Transactions Retry Example 3
+
+      def run_transaction_with_retry(session)
+        loop do
+          begin
+            yield session # performs transaction
+            break
+          rescue Mongo::Error::OperationFailure, Mongo::Error::SocketError,
+                 Mongo::Error::NoServerAvailable => e
+
+            puts 'Transaction aborted. Caught exception during transaction.'
+            raise unless e.label?(Mongo::Session::TRANSIENT_TRANSACTION_ERROR_LABEL)
+
+            puts "#{Mongo::Session::TRANSIENT_TRANSACTION_ERROR_LABEL}, retrying transaction ..."
+            next
```

Comment:
`next` is unneeded here.

Reply:
Given that the sample code explicitly uses `continue` (as do the already-published examples for other languages), I opted to put it here as well

Reply:
It's not necessary there either.